templates/includes/records/hierarchy-local.html

```diff
@@ -29,7 +35,13 @@
                     <p class="hierarchy-local__nav-item">
                         <span class="hierarchy-local__nav-indicator hierarchy-local__nav-indicator--right"></span>
                         <span class="hierarchy-local__nav-label">{{ 'next_record'|as_label }}</span>
-                        <a href="{% record_url page.next_record %}" data-link="Next" class="link--bold"><span class="sr-only">{{ 'next_record'|as_label }}</span> {{ page.next_record.iaid }}</a>
+                        <a href="{% record_url page.next_record %}" data-link="Next" class="link--bold">
+                            {% if page.next_record.has_reference_number %}
+                                <span class="sr-only">{{ 'next_record'|as_label }}</span> {{ page.next_record.reference_number }}
+                            {% else %}
+                                Next {{ page.level.lower|default:"record" }}
```

Comment:
Similar to Previous comment - Next label is already present above `{{ 'next_record'|as_label }}`

Reply:
It's important that the 'Next' text is part of the link, as that will be read out by screen readers